oemof/energy_system.py

```diff
@@ -36,7 +36,9 @@ class EnergySystem:
         Stored in the :attr:`entities` attribute.
         Defaults to `[]` if not supplied.
     timeindex : pandas.datetimeindex
-        Define the time range and increment for the energy system.
+        Define the time range for the energy system.
+    timeincrement : numeric (sequence)
```

Comment:
As this PR is filed against 0.3.3, the API should be stable. So, I assume it is still possible to not give a timeincrement. This should be documented better: What would be the default?

Reply:
If no timeincrement was given, it will be calculated regularly as in former versions. I don't think passing a default timeincrement is a good idea. Right now (according to my changes), there are two possibilities:
1. giving a timeincrement and timeindex: everything is fine, timeincrement will be used for further calculations.
2. giving a timeindex: timeindex has to be valid, it needs to have a 'freq' attribute -> constant timeincrement will be calculated as usual

Reply:
There is no default. If the `timeincrement` is `None` it will be calculated from the time index. This works only with an equidistant time index. The problem is, that the unit of the calculated `timeincrement` is hour even though we normally do not use units. It is possible to use Joule instead of kWh but in that case the `timeincrement` has to be in seconds and the loss rate should be the rate per second.

Reply:
The PR is now against dev, so API changes are in order.